android/src/app/organicmaps/bookmarks/BookmarkListAdapter.java

```diff
@@ -513,4 +518,27 @@ public Object getItem(int position)
       return info;
     }
   }
+
+  private void setMoreButtonVisibility(TextView text, TextView moreBtn)
+  {
+    text.post(() ->
+    {
+      final int lineCount = text.getLineCount();
+      if (lineCount > 2)
+      {
+        text.setMaxLines(2);
+        moreBtn.setVisibility(View.VISIBLE);
+      }
+      else
+        moreBtn.setVisibility(View.GONE);
+    });
+  }
+
+  private void onMoreButtonClicked(TextView moreBtn, TextView text)
+  {
+    if (text.getMaxLines() == 2)
+      text.setMaxLines(Integer.MAX_VALUE);
+    else
+      text.setMaxLines(2);
```

Comment:
We have "2" four time in this module. Should take out as a separate constant inside Adapter class.